fundingmanager.go

```diff
@@ -176,6 +177,25 @@ type numPendingReq struct {
 	resp chan uint32
 }
 
+// TODO: (andrew.shvv) Find better place for this function.
+// checkMaxPending check number of pending channels to not exceed maximum value
+// send the lnwire.ErrorGeneric message to the offending peer if the constraint
+// is violated.
+func (f *fundingManager) checkMaxPending(peer *peer) bool {
+	if len(f.activeReservations[peer.id]) >= maxActiveReservations {
+		errMsg := &lnwire.ErrorGeneric{
+			Problem: "Number of pending channels exceed maximum",
+			ErrorID: lnwire.ErrorMaxPendingChannels,
+		}
+
+		peer.queueMsg(errMsg, nil)
```

Comment:
The observance of error messages hasn't yet been fleshed out within the `peer` code. Depending on the severity of the error, if the error specifically effects an open channel, then the `channelManager` should initiate a unilateral closure of the channel by broadcasting the latest state. However, in many cases like this one, no such action needs to be carried out, only reporting the error is sufficient.